git-bash-hints.qmd

```diff
@@ -0,0 +1,310 @@
+
+::: callout-important
+Make sure you're in the correct directory/using the correct path before using commands in this section.
+:::
+
+`rm`: This stands for "remove" and we can use it to remove files and directories.
+
+To remove a directory and its contents, we would amend the code below to reflect the path and the directory we want and enter it into the terminal:
+
+:::{.callout-warning}
+
+## Using recursive mode 
+
+Be cautious when using the rm -r command in Git Bash, as it recursively deletes the specified directory and all its contents. This action is irreversible, so double-check the directory path to avoid accidentally deleting important files or directories.
+:::
+
+```{bash }
+
+rm -r path_to_the_directory/directory_name
+
+```
+
+### Removing files
+
+Assuming we're already in the directory we want, we can remove a file from it by amending the code below to reflect the file name we want to remove, and its extension and then enter it into the terminal:
+
+```{bash }
+
+rm file_name.extention
+
+```
+
+You need to make sure to include the extension of the file in the command, or it will not work.
+
+## Viewing File Contents
+
+::: callout-important
+Make sure you're in the correct directory/using the correct path before using commands in this section.
+:::
+
+The commands in this section allow you to display the contents of files, either in full or in parts, for easy reading and inspection.
+
+To display the first 10 lines of a file, we would amend the code below to reflect the file name we want to view and its extension and then enter it into the terminal:
+
+```{bash }
+
+head file_name.extention
+
+```
+
+You can also customise the number of lines you want to view by adding `-number of lines` after `head`. For example, if we want to view the first 12 lines of a file, we use: 
+
+```{bash }
+
+head -12 file_name.extention
+
+```
+
+To display the bottom 10 lines of a file, we would amend the code below to reflect the file name we want to view and its extension and then enter it into the terminal:
+
+```{bash }
+
+tail file_name.extention
+
+```
+
+You can also customise the number of lines you want to view by adding `-number of lines` after `tail`. For example, if we want to view the last 3 lines of a file, we use: 
+
+```{bash }
+
+tail -3 file_name.extention
+
+```
+
+To display the contents of a file, we would amend the code below to reflect the file name we want to view and its extension and then enter it into the terminal:
+
+```{bash }
+
+cat file_name.extention
+
+```
+
+## Searching and Finding Files
+
+### Finding files by patterns
+
+::: callout-important
+Make sure you're in the correct directory/using the correct path before using commands in this section.
+:::
+
+You can look for files in your repository that match a certain pattern by using `git grep`. We would amend the code below by replacing `"pattern"` with the term or pattern we want to search for in files.
+
+```{bash }
+
+git grep "pattern"
```

Comment:
Interesting, I didn't realise Git had their own implementation of grep. I think stick to the Bash one here and we can maybe have the Git version as an alternative in the Git glossary somewhere.
```suggestion
grep "pattern" filename
```